### ISSUE TITLE
Wish: Template capability

### ISSUE THREAD

**Issue by andydandy74** [DynamoDS/Dynamo#6143]
It would be great if we could specify a *.dyn template file that is opened each time a user creates a new workflow. This would help a lot with maintaining graph annotation standards across a multi-user environment.

**Comment by josols** [comment]
+1

**Comment by radumg** [comment]
+1. Apart from standards consistency in a practice environment, it would also help get new users up and running quicker. We already have Grasshopper templates for example, so we'd be looking to align them. 
Is this being worked on @Racel ?

**Comment by Racel** [comment]
@radumg - Not exactly. This particular feature is not in our short term roadmap, but we may be able to address the "reason" for this wish in a different way in the next implementation of Dynamo that we are working on.

@andydandy74 @radumg @eibre - what kind of things are you looking to add to this template and what do you want to accomplish by using this template? Do you want to make sure that all inputs are colored or grouped a certain way? Do you want to ensure names of nodes are consistent?

**Comment by andydandy74** [comment]
@Racel - Our template contains a basic graph layout that illustrates all the rules of our style guide. The goal is for every graph to adhere to those rules so that users will find it easier to interact with the graphs (e.g. group colours based on functionality, positions of input and output elements, documentation style etc.).

**Comment by radumg** [comment]
@Racel , yes to your questions above (colouring, grouping) and i'll second what @andydandy74 said above. To illustrate, here's a few examples.

This one is by Mark (now DesignTech.io) and what we use at the office.
![designtech](https://cloud.githubusercontent.com/assets/11439624/23188982/b414475e-f888-11e6-9010-0a1b92625a45.jpg)
source : http://designtech.io/tools/door-renumbering/

Another great example is the post by White Arkitekten. The standard they shared is very popular and I myself used it at a previous practice.
![white](https://cloud.githubusercontent.com/assets/11439624/23188988/b5d42fd2-f888-11e6-9807-bb3db9b0dfb0.jpg)
source : http://dynamobim.org/dynamo-graphic-standards-at-white-arkitekter-ab/

**Comment by Racel** [comment]
Thanks @andydandy74 and @radumg - That is helpful....

So, all we need is the following:
- A way to specify the file path to the template dyn
- A way to enable/disable the template (should this happen on new file creation or as a global setting?)

After talking with @mjkkirschner - This seems pretty straight forward. Will file a task for this and we will figure out how to get it in after 1.3.

**Comment by andydandy74** [comment]
Perhaps the easiest soluton would be to have it in DynamoSettings.xml only:
- Use a template if a path is specified, otherwise use the default workspace
- No need to expose this in the UI since in most cases a template will be used in deployments and users should not be able to disable it effortlessly

**Comment by radumg** [comment]
what @andydandy74 said 👍

**Comment by Racel** [comment]
Thanks @andydandy74 and @radumg - That makes this a bit easier.

**Comment by ksobon** [comment]
Also, Python editor template would be nice...:-) just saying what these guys are saying. https://github.com/DynamoDS/Dynamo/issues/7604

**Comment by smangarole** [comment]
@Racel can you please comment on these enhancement requests? @Dewb @kronz FYI

**Comment by radumg** [comment]
@smangarole FYI, Python editor template is awaiting team feedback in PR #8034 and I can then finish implementing it. Also planning to add DYN template support after that using similar mechanisms.

**Comment by smangarole** [comment]
@radumg Thanks for the reply. We have your PR in review/testing. Will send you an update once it's merged. Thanks again for the PR.

**Comment by radumg** [comment]
Thanks @smangarole - that PR will be changing tonight as i incorporate feedback, will give you a mention there when it's done.

**Comment by radumg** [comment]
@smangarole , please check out PR #8122 for the updated version of the PR. cheers!

**Comment by ondejcik** [comment]
Any updates to DYN template ?? i dont see it implemented in 2.0.1

**Comment by radumg** [comment]
Hey @ondejcik , made a prototype of this back in Feb for the extensions workshop but unfortunately I haven’t had time to finish it since then 😔
Might try and pick it up soon 😎 but can’t promise anything 😇

**Comment by andydandy74** [comment]
FYI: You can try out the DynaMonito view extension. It adds template capability to Dynamo. https://github.com/andydandy74/Monito/releases
